qiskit_nature/second_q/drivers/psi4d/psi4driver.py

```diff
@@ -0,0 +1,247 @@
+
+        name = "".join(molecule.symbols)
+        geom = "\n".join(
+            [
+                name + " " + " ".join(map(str, coord))
+                for (name, coord) in zip(molecule.symbols, molecule.coords)
+            ]
+        )
+        cfg1 = f"molecule {name} {{\nunits {units}\n"
+        cfg2 = f"{molecule.charge} {molecule.multiplicity}\n"
+        cfg3 = f"{geom}\nno_com\nno_reorient\n}}\n\n"
+        cfg4 = f"set {{\n basis {basis}\n scf_type pk\n reference {method.value}\n}}"
+        return Psi4Driver(cfg1 + cfg2 + cfg3 + cfg4)
+
+    @staticmethod
+    def to_driver_basis(basis: str) -> str:
+        """
+        Converts basis to a driver acceptable basis
+        Args:
+            basis: The basis set to be used
+        Returns:
+            driver acceptable basis
+        """
+        if basis == "sto3g":
+            return "sto-3g"
+        return basis
+
+    @staticmethod
+    def check_method_supported(method: MethodType) -> None:
+        """
+        Checks that PSI4 supports this method.
+        Args:
+            method: Method type
+        Raises:
+            UnsupportMethodError: If method not supported.
+        """
+        if method not in [MethodType.RHF, MethodType.ROHF, MethodType.UHF]:
+            raise UnsupportMethodError(f"Invalid PSI4 method {method.value}.")
+
+    def run(self) -> ElectronicStructureProblem:
+        cfg = self._config
+
+        psi4d_directory = Path(__file__).resolve().parent
+        template_file = psi4d_directory.joinpath("_template.txt")
+
+        input_text = [cfg]
+
+        file_fd, hdf5_file = tempfile.mkstemp(suffix=".hdf5")
+        os.close(file_fd)
+        input_text += [f'_FILE_PATH = "{Path(hdf5_file).as_posix()}"']
+
+        with open(template_file, "r", encoding="utf8") as file:
+            input_text += [line.strip("\n") for line in file.readlines()]
+
+        file_fd, input_file = tempfile.mkstemp(suffix=".inp")
+        os.close(file_fd)
+        with open(input_file, "w", encoding="utf8") as stream:
+            stream.write("\n".join(input_text))
+
+        file_fd, output_file = tempfile.mkstemp(suffix=".out")
+        os.close(file_fd)
+        try:
+            Psi4Driver._run_psi4(input_file, output_file)
+            if logger.isEnabledFor(logging.DEBUG):
+                with open(output_file, "r", encoding="utf8") as file:
+                    logger.debug("Psi4 output file:\n%s", file.read())
+        finally:
+            run_directory = os.getcwd()
+            for local_file in os.listdir(run_directory):
+                if local_file.endswith(".clean"):
+                    os.remove(run_directory + "/" + local_file)
+            try:
+                os.remove("timer.dat")
+            except Exception:  # pylint: disable=broad-except
+                pass
+
+            try:
+                os.remove(input_file)
+            except Exception:  # pylint: disable=broad-except
+                pass
```

Comment:
Is there a better way than catching an exception and doing nothing? Maybe one could check if the file exists?

Reply:
Once we can drop Python 3.7 support, we will be able to use the `missing_ok` argument of `pathlib.Path.unlink`. Until then, this can be left as is